jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -647,4 +656,29 @@ private void writeTarFile(
   private <E> List<Future<E>> scheduleCallables(ImmutableList<? extends Callable<E>> callables) {
     return callables.stream().map(executorService::submit).collect(Collectors.toList());
   }
+
+  private String computeArchitecture(String architecture) {
+    if (architecture.equals("x86_64")) {
+      return "amd64";
+    } else if (architecture.equals("aarch64")) {
+      return "arm64";
+    }
+    return architecture;
+  }
+
+  private Optional<Image> fetchBuiltImageForLocalBuild(String osType, String architecture)
+      throws InterruptedException, ExecutionException {
+    if (results.baseImagesAndBuiltImages.get().size() > 1) {
+      LOGGER.warning(
+          "Detected multi-platform configuration, only building the one that matches the local Docker Engine's os and architecture");
+    }
+    for (Map.Entry<Image, Future<Image>> imageEntry :
```

Comment:
I see that they are already tested in `JibIntegrationTest`, but ideally I think the different cases should be tested in `StepsRunnerTest` since we don't have to run docker engine. I'll leave it to you to decide if the refactoring it needed.